packages/core/src/model.js

```diff
@@ -3188,9 +3188,16 @@ Instead of specifying a Model, either:
       || options.clone
     ) {
       const values = Object.create(null);
+      const modelDefinition = this.constructor.modelDefinition;
       if (attributesWithGetters.size > 0) {
         for (const attributeName2 of attributesWithGetters) {
-          if (!this._options.attributes?.includes(attributeName2)) {
+          const attributeDefinition = modelDefinition.rawAttributes[attributeName2];
```

Comment:
rawAttributes is used to write attributes. It's the state of the attribute before being normalized. These attributes are then normalized and put in `attributes`, which is what you need to use. It's already available in this method